### PR TITLE
Enable usage of syn from 'panic = abort' code

### PR DESCRIPTION
**Pull request by smklein**
Refer to the inline code comment for all the details.

### CODE REVIEW

File: dropshot/src/router.rs
Comment (ahl) [inline dropshot/src/router.rs:190]: FWIW @davepacheco has tried (and failed) to hold the line on /* */ comments in dropshot so... do what you will with that .

**Review by ahl — APPROVED** [review]
Nice work chasing this down? Is there anything related to the initial bug beyond what you document in the code that's worth capturing in this PR?

File: dropshot/src/router.rs
Comment (smklein) [reply dropshot/src/router.rs:190]: Interesting, I always considered that comment style "valid, but non-idiomatic" - the Rust book only shows the "//" style (see: [here](https://doc.rust-lang.org/stable/book/ch03-04-comments.html))  and there was even an (admittedly unsuccessful) attempt to remove block comments: https://github.com/rust-lang/rfcs/issues/1371

(Plus I [heard a rustlang maintainer also had the spicy take that they're not idiomatic](https://github.com/rust-lang/book/issues/693#issuecomment-301200629))

Do you want me to switch before landing? I noticed we're pretty inconsistent across many of Oxide's repos (dunno if we should clarify which repos do / don't have this styling more explicitly).

Comment (smklein) [reply dropshot/src/router.rs:190]: [The rustdoc book](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html) also exclusively uses the non-block comment style ("//" or "///")

**Comment by smklein** [comment]
> Nice work chasing this down? Is there anything related to the initial bug beyond what you document in the code that's worth capturing in this PR?

I don't believe so - it's a bit hairy, and might change if we made other calls to `syn`/`quote` from this codebase, but we don't really have a better "initialization point" that wouldn't push this onto dropshot users.

I'm kinda hoping that this gets resolved in upstream rust, fixed in proc_macro2, we can remove the hack, and we'll be done with it.

File: dropshot/src/router.rs
Comment (ahl) [reply dropshot/src/router.rs:190]: I know.

**Pull request merged**
